src/game/server/entities/laser.cpp

```diff
@@ -43,11 +43,50 @@ void CLaser::DoBounce()
 		GameWorld()->DestroyEntity(this);
 		return;
 	}
+	int CheckFor;
+	float Elasticity;
+	float EnergyMultiplier = 1;
+	if (GameServer()->Collision()->TestBox(m_Pos, vec2(0.0f, 0.0f), 8))
+	{
+		CheckFor = 0;
+		EnergyMultiplier = GameServer()->Tuning()->m_LaserWaterReachMultiplier;
+		Elasticity = -(1 / GameServer()->Tuning()->m_LaserWaterDiffraction);
+	}
+	else
+	{
+		CheckFor = 8;
+		EnergyMultiplier = 1;
+		Elasticity = -GameServer()->Tuning()->m_LaserWaterDiffraction;
+	}
+	vec2 To = m_Pos + m_Dir * (m_Energy / EnergyMultiplier);
+	int Result = GameServer()->Collision()->IntersectLineWithWater(m_Pos, To, 0x0, &To, CheckFor);
+	if(Result == 2)
+	{
+		if (!HitCharacter(m_Pos, To))
+		{
+			// intersected
+			m_From = m_Pos;
+			m_Pos = To;
+
+			vec2 TempPos = m_Pos;
+			vec2 TempDir = m_Dir * 4.0f;
```

Comment:
@Stiopa866 what is the point in making this vector longer here?

Reply:
I am unsure, the multiplication is present in the usual code.